superset/models/helpers.py

```diff
@@ -1597,6 +1593,10 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                     template_processor=template_processor
                 )
                 need_groupby = True
+            elif col in columns_by_name:
+                col = self.convert_tbl_column_to_sqla_col(
+                    columns_by_name[col], template_processor=template_processor
+                )
```

Comment:
I'm basically just changing the order to first evaluate the column against metrics, and then across columns. 

My understanding of the issue is that in the current `if/elif` ordering, we would first evaluate `col` across `columns_by_name`, which would be true (even tho it's a metric), and then `convert_tbl_column_to_sqla_col()`  does `col = sa.column(tbl_column.column_name, type_=type_)` which seems to return quoted if the column is uppercase (hence the issue does not happen with `normalize_columns` disabled):

``` python
import sqlalchemy as sa
print(sa.column("test", sa.String))
print(sa.column("TEST", sa.String))
```

``` bash
test
"TEST"
```

With `normalize_columns`, we send a lowercase column label to `sa.column()` which then does not quote it.